src/csharp/Microsoft.Spark.E2ETest/UdfTests/UdfSimpleTypesTests.cs

```diff
@@ -21,10 +22,28 @@ public class UdfSimpleTypesTests
         public UdfSimpleTypesTests(SparkFixture fixture)
         {
             _spark = fixture.Spark;
-            _df = _spark
-                .Read()
-                .Schema("name STRING, age INT, date DATE")
-                .Json(Path.Combine($"{TestEnvironment.ResourceDirectory}people.json"));
+            var data = new List<GenericRow>();
+            data.Add(new GenericRow(
+                new object[]
+                {
+                    null,
+                    new Date(2020, 1, 1),
+                    new Timestamp(2020, 1, 1, 0, 0, 0, 0)
+                }));
+            data.Add(new GenericRow(
+                new object[]
+                {
+                    30,
+                    new Date(2020, 1, 2),
+                    new Timestamp(2020, 1, 2, 15, 30, 30, 0)
```

Comment:
Can we use non zero value for microseconds?

Reply:
Got it, good catch! So actually non zero microseconds doesn't work for udf, it always converts to zero microseconds when reading input data [here](https://github.com/dotnet/spark/blob/9275678a6e46e06a266e5516b3fea50896ede761/src/csharp/Microsoft.Spark.Worker/Command/SqlCommandExecutor.cs#L135).